mypy/checkexpr.py

```diff
@@ -3918,6 +3918,10 @@ def visit_assert_type_expr(self, expr: AssertTypeExpr) -> Type:
             always_allow_any=True,
         )
         target_type = expr.type
+        if isinstance(mypy.types.get_proper_type(source_type), mypy.types.Instance) and (
+            hasattr(source_type, "last_known_value") and source_type.last_known_value is not None
+        ):
+            source_type = source_type.last_known_value
```

Comment:
Thanks! I think this is close to right, just needs a quick tweak

Effectively, the reason why we're doing `isinstance(get_proper_type(source_type), mypy.types.Instance)` is because we want to check for the `last_known_value` attribute.

The difference between `source_type` and `get_proper_type(source_type)` is that if `source_type` is a type alias, `get_proper_type(source_type)` will be the actual type we care about.

```suggestion
        proper_source_type = get_proper_type(source_type)
        if isinstance(proper_source_type, mypy.types.Instance) and proper_source_type.last_known_value is not None:
            source_type = proper_source_type.last_known_value
```